### PR TITLE
Espresso improvements

### PR DESCRIPTION
**Pull request by ghostbuster91**

### CODE REVIEW

File: espresso/src/main/java/com/elpassion/android/commons/espresso/matchers/TextInputEditTextHintMatcher.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    override fun matchesSafely(view: TextInputEditText): Boolean {
+        val grandpa = view.parent?.parent
+        if (grandpa is TextInputLayout) {
+            return grandpa.hint == view.context.getString(hintStringId)
+        }
+        return false
+    }
+
+    override fun describeTo(description: Description) {
+        description.appendText("has hint text from string resource on TextInputLayout: " + hintStringId)
```
Comment (jozzi05) [inline espresso/src/main/java/com/elpassion/android/commons/espresso/matchers/TextInputEditTextHintMatcher.kt]: Use kotlin syntax

File: espresso/src/main/java/com/elpassion/android/commons/espresso/matchers/EditTextErrorMatcher.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    override fun matchesSafely(view: EditText): Boolean {
+        val grandpa = view.parent?.parent
+        if (grandpa is TextInputLayout) {
+            return grandpa.error == view.context.getString(textId)
+        }
+        return false
+    }
+
+    override fun describeTo(description: Description) {
+        description.appendText("has error text from string resource on TextInputLayout: " + textId)
```
Comment (jozzi05) [inline espresso/src/main/java/com/elpassion/android/commons/espresso/matchers/EditTextErrorMatcher.kt]: Use kotlin syntax.

File: espresso/src/main/java/com/elpassion/android/commons/espresso/matchers/EditTextErrorMatcher.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package com.elpassion.android.commons.espresso.matchers
+
+import android.support.annotation.StringRes
+import android.support.design.widget.TextInputLayout
+import android.support.test.espresso.matcher.BoundedMatcher
+import android.view.View
+import android.widget.EditText
+import org.hamcrest.Description
+
+class EditTextErrorMatcher(@StringRes private val textId: Int) : BoundedMatcher<View, EditText>(EditText::class.java) {
```
Comment (jozzi05) [inline espresso/src/main/java/com/elpassion/android/commons/espresso/matchers/EditTextErrorMatcher.kt:10]: Change `textId ` to `errorrStringId` for consistency or change TextInputEditTextHintMatcher field to `textId`.

**Review by jozzi05 — CHANGES_REQUESTED** [review]
It would be nice to also have: 
```kotlin
fun checkIntent(clazz: Class<out Activity>) {
      Intents.intended(IntentMatchers.hasComponent(clazz.name))
}
```

File: espresso/src/main/java/com/elpassion/android/commons/espresso/matchers/TextInputEditTextHintMatcher.kt (hunk 1 above)
Comment (jozzi05) [reply espresso/src/main/java/com/elpassion/android/commons/espresso/matchers/TextInputEditTextHintMatcher.kt]:
```kotlin
"...$hintStringId"
```

**Review by marcinsus — APPROVED** [review]

**Review by jozzi05 — APPROVED** [review]

**Pull request merged**